cpp/src/arrow/ipc/message.cc

```diff
@@ -275,9 +275,7 @@ Status ReadMessage(int64_t offset, int32_t metadata_length, io::RandomAccessFile
   }
 
   if (flatbuffer_length == 0) {
-    // EOS
-    *message = nullptr;
-    return Status::OK();
+    return Status::Invalid("Unexpected empty message in IPC file format");
```

Comment:
@wesm Is this correct? AFAIU, even if an IPC file has an EOS message at the end, it should never be read using `ReadMessage`.

Reply:
Right, I think this is fine if there is no code assuming that they get a null message back (we'd probably want to fix such code anyway based on this experience)